torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,416 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+def download(url, raw_folder, dataset_name):
```

Comment:
```
extract_archive(
    download_from_url(
        torchtext.datasets.supervised.urls['AG_NEWS']) #returns a path in cache, 
        "path/to/AG_NEWS/folder"
    )
)
```

Reply:
I don't this necessary needs to be its own function

Reply:
minor change. Put download and extract_archive together.